projects/organization-management/src/app/models/node/node.mapper.spec.ts

```diff
@@ -0,0 +1,270 @@
+
+import { ResourceIdentifierData } from '../resource-identifier/resource-identifier.interface';
+
+import { NodeData, NodeListDocument } from './node.interface';
+import { NodeMapper } from './node.mapper';
+import { Node } from './node.model';
+
+const OILCORP_GERMANY: NodeData = {
+  id: 'OilCorp_Germany',
+  attributes: {
+    description: 'The German division of Oil Corp.',
+    name: 'Oil Corp Germany',
+  },
+  relationships: {
+    organization: {
+      data: {
+        id: 'oilcorp.example.org',
+      },
+    },
+    childNodes: {
+      data: [{ id: 'OilCorp_Berlin' }, { id: 'OilCorp_Jena' }],
+    },
+    parentNode: { data: undefined },
+  },
+};
+
+const OILCORP_BERLIN: NodeData = {
+  id: 'OilCorp_Berlin',
+  attributes: {
+    description: 'The Berlin headquarter of Oil Corp.',
+    name: 'Oil Corp Berlin',
+  },
+  relationships: {
+    organization: {
+      data: {
+        id: 'oilcorp.example.org',
+      },
+    },
+    childNodes: undefined,
+    parentNode: {
+      data: {
+        id: 'OilCorp_Germany',
+      },
+    },
+  },
+};
+
+describe('Node Mapper', () => {
+  let nodeMapper: NodeMapper;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    nodeMapper = TestBed.inject(NodeMapper);
+  });
+
+  describe('fromDocument', () => {
+    it('should throw when input is falsy', () => {
+      expect(() => nodeMapper.fromDocument(undefined)).toThrow();
+    });
+
+    it('should map incoming no data to empty object', () => {
+      const data = { data: [] } as NodeListDocument;
+      const mapped = nodeMapper.fromDocument(data);
+      expect(mapped).toBeTruthy();
+      expect(mapped.edges).toBeEmpty();
+      expect(mapped.rootIds).toBeEmpty();
+      expect(mapped.nodes).toBeEmpty();
+    });
+
+    it('should map incoming data to model data', () => {
+      const data = { data: [OILCORP_GERMANY, OILCORP_BERLIN] } as NodeListDocument;
+      const mapped = nodeMapper.fromDocument(data);
+      expect(mapped.rootIds).toEqual(['OilCorp_Germany']);
+      expect(mapped.edges.OilCorp_Germany).toEqual(['OilCorp_Berlin', 'OilCorp_Jena']);
+    });
+  });
+
+  describe('fromData', () => {
+    it('should throw when input is falsy', () => {
+      expect(() => nodeMapper.fromData(undefined)).toThrow();
+    });
+
+    it('should map incoming data to model data', () => {
+      const data = OILCORP_BERLIN;
+      const mapped = nodeMapper.fromData(data);
+
+      expect(mapped).toHaveProperty('nodes.OilCorp_Berlin');
+      const mappedElement = mapped.nodes.OilCorp_Berlin;
+      expect(mappedElement).toHaveProperty('id', 'OilCorp_Berlin');
+      expect(mappedElement).toHaveProperty('name', 'Oil Corp Berlin');
+      expect(mappedElement).toHaveProperty('description', 'The Berlin headquarter of Oil Corp.');
+      expect(mappedElement).toHaveProperty('organization', 'oilcorp.example.org');
+      expect(mapped.edges).toBeEmpty();
+      expect(mapped.rootIds).toBeEmpty();
+    });
```

Comment:
I guess we currently prefer `toMatchInlineSnapshot` <-- so much more convenient. 😉